File: NirvanaSMS.Api.CSharp/Operations/BlackListOperations.cs
using NirvanaSMS.Api.Core.Base;
using NirvanaSMS.Api.Core.Entity;
using NirvanaSMS.Api.Core.Utility;

namespace NirvanaSMS.Api.Operations
{
    public class BlackListOperations : BlackListBase
    {
        public override ProcessResult Add(ApiUser apiUser, PhoneNumber phoneNumber)
        {
            PhoneNumberFormatter formatter = new PhoneNumberFormatter();
            PhoneNumber formattedNumber = formatter.Format(phoneNumber);
            return base.Add(apiUser,formattedNumber);
        }
    }
}

File: NirvanaSMS.Api.Java/src/main/java/org/nirvanasms/api/operations/BlackListOperations.java
package org.nirvanasms.api.operations;

import org.nirvanasms.api.core.base.BlackListBase;
import org.nirvanasms.api.core.entity.ApiUser;
import org.nirvanasms.api.core.entity.PhoneNumber;
import org.nirvanasms.api.core.exceptions.NonValidPhoneNumberException;
import org.nirvanasms.api.core.utility.PhoneNumberFormatter;
import org.nirvanasms.api.core.utility.ProcessResult;

public class BlackListOperations extends BlackListBase {
	@Override
	public ProcessResult<?> add(ApiUser apiUser, PhoneNumber phoneNumber) throws NonValidPhoneNumberException{
		 PhoneNumberFormatter formatter = new PhoneNumberFormatter();
         PhoneNumber formattedNumber = formatter.format(phoneNumber);
         return super.add(apiUser,formattedNumber);
	}
}
